framework/src/base/ComputeJacobianThread.C

```diff
@@ -179,6 +152,18 @@ ComputeJacobianThread::subdomainChanged()
   _fe_problem.setActiveElementalMooseVariables(needed_moose_vars, _tid);
   _fe_problem.setActiveMaterialProperties(needed_mat_props, _tid);
   _fe_problem.prepareMaterials(_subdomain, _tid);
+
+  // If users pass a empty vector or a full size of vector,
+  // we take all kernels
+  if (!_tags.size() || _tags.size() == _fe_problem.numVectorTags())
```

Comment:
This whole `if` statement is a good idea!  Way to think about optimizing the usual case (all or 1) 👍